amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -51,72 +41,88 @@ public void testMultiStreamDeserializationSuccess() {
     @Test
     public void testMultiStreamDeserializationFail() {
         for (final String pattern : Arrays.asList(
-                // arn examples
-                "arn:aws:kinesis::123456789012:stream/stream-name", // missing region
-                "arn:aws:kinesis:region::stream/stream-name", // missing account id
-                "arn:aws:kinesis:region:123456789:stream/stream-name", // account id not 12 digits
-                "arn:aws:kinesis:region:123456789abc:stream/stream-name", // 12char alphanumeric account id
-                "arn:aws:kinesis:region:123456789012:stream/", // missing stream-name
-                // serialization examples
                 ":stream-name:123", // missing account id
 //                "123456789:stream-name:123", // account id not 12 digits
                 "123456789abc:stream-name:123", // 12char alphanumeric account id
                 "123456789012::123", // missing stream name
                 "123456789012:stream-name", // missing delimiter and creation epoch
                 "123456789012:stream-name:", // missing creation epoch
+                "123456789012:stream-name:-123", // negative creation epoch
                 "123456789012:stream-name:abc", // non-numeric creation epoch
                 ""
         )) {
             try {
                 StreamIdentifier.multiStreamInstance(pattern);
-                Assert.fail(pattern + " should not have created a StreamIdentifier");
+                Assert.fail("Serialization " + pattern + " should not have created a StreamIdentifier");
             } catch (final IllegalArgumentException iae) {
                 // expected; ignore
             }
         }
     }
 
+    /**
+     * Test Arns that <b>should not</b> match a valid AWS Kinesis stream Arn.
+     */
     @Test
-    public void testInstanceFromArn() {
-        final Arn arn = toArn(KINESIS_REGION);
-        final StreamIdentifier single = StreamIdentifier.singleStreamInstance(arn.toString());
-        final StreamIdentifier multi = StreamIdentifier.multiStreamInstance(arn.toString());
-
-        assertEquals(single, multi);
-        assertEquals(Optional.empty(), single.streamCreationEpochOptional());
-        assertActualStreamIdentifierExpected(arn, single);
+    public void testMultiStreamByArnWithInvalidStreamArnFail() {
+        final String region = KINESIS_REGION.id();
```

Comment:
[Primitive obsession](https://wiki.c2.com/?PrimitiveObsession). Pass a `Region` into `streamArn(...)` and let *it* Stringify (if non-null).